src/components/map/layers/NetworkLayers.tsx

```diff
@@ -62,39 +68,65 @@ export default class NetworkLayers extends Component<INetworkLayersProps> {
         };
     }
 
-    private getPointStyle = () => {
+    private isLinkHidden = (transitType: TransitType, startNodeId: string, endNodeId: string) => {
+        const selectedTransitTypes = this.props.networkStore!.selectedTransitTypes;
+        if (!selectedTransitTypes.includes(transitType)) {
+            return true;
+        }
+        const node = this.props.editNetworkStore!.node;
+        if (node && (node.id === startNodeId || node.id === endNodeId)) {
+            return true;
+        }
+        return false;
+    }
+
+    private getLinkPointStyle = () => {
         return {
             // Layer name 'piste' is directly mirrored from Jore through geoserver
             piste: (properties: any, zoom: number) => {
-                const selectedTransitTypes = this.props.networkStore!.selectedTransitTypes;
+                const {
+                    lnkverkko: transitTypeCode,
+                    lnkalkusolmu: startNodeId,
+                    lnkloppusolmu: endNodeId,
+                } = properties;
                 const transitType = TransitTypeHelper
-                    .convertTransitTypeCodeToTransitType(properties.lnkverkko);
-                const color = TransitTypeColorHelper.getColor(transitType);
+                    .convertTransitTypeCodeToTransitType(transitTypeCode);
 
-                if (!selectedTransitTypes.includes(transitType)) {
+                if (this.isLinkPointHidden(transitType, startNodeId, endNodeId)) {
                     return this.getEmptyStyle();
                 }
 
                 return {
-                    color,
+                    color: TransitTypeColorHelper.getColor(transitType),
                     radius: 1,
                 };
             },
         };
     }
 
+    private isLinkPointHidden = (transitType: TransitType, startNodeId: string, endNodeId: string) => { // tslint:disable max-line-length
```

Comment:
This method looks identical to isLinkHidden()

Reply:
I was thinking this too but decided to have them in their own method (their logic might change later). But maybe its better to use a common method for now -> refactored.